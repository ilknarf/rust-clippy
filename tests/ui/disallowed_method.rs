#![warn(clippy::disallowed_method)]
#![allow(clippy::no_effect)]

struct Foo;

impl Foo {
    fn bad_method(self) {}
}

struct ImplStruct;

trait Baz {
    fn bad_method(self);
}

impl Baz for ImplStruct {
    fn bad_method(self) {}
}

struct NormalStruct;

impl NormalStruct {
    fn bad_method(self) {}
}

struct AttrStruct {
    bad_method: i32,
}

fn main() {
    let f = Foo;
    let b = ImplStruct;
    let n = NormalStruct;
    let a = AttrStruct{ bad_method: 5 };

    // lint these
    f.bad_method();
    b.bad_method();
    // these are good
    n.bad_method();
    a.bad_method;
}
