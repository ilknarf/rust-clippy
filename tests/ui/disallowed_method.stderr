error: Use of a disallowed trait method `bad_method` of `disallowed_method::Baz`
  --> $DIR/disallowed_method.rs:48:5
   |
LL |     b.bad_method();
   |     ^^^^^^^^^^^^^^
   |
   = note: `-D clippy::disallowed-method` implied by `-D warnings`

error: Use of a disallowed trait method `bad_method` of `disallowed_method::Baz`
  --> $DIR/disallowed_method.rs:49:5
   |
LL |     c.bad_method();
   |     ^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

